### DISCUSSION TITLE
How about a maintenance mode extension?

### DISCUSSION THREAD

**Discussion by pftnhr** [datenstrom/community#781] [Ask a question] [answered]
I would like to have an extension to put the whole website into a maintenance mode so that I am not working on the website in front of everyone. Unfortunately, my coding skills are far from sufficient to implement something like that. As a work around, a .htaccess would certainly work, but that would somehow be too dirty for such a relaxed CMS.

**Comment by schulle4u** [comment, marked as answer]
There is a [maintenantce extension](https://github.com/nogginfuel/yellow-plugin-maintenance), but it hasn't been updated for years and won't work with the latest API. @nogginfuel are you still maintaining this project? 

Basically you can have a look at the [draft extension](https://github.com/datenstrom/yellow-extensions/tree/master/source/draft), it should be quite simple to modify it for page/site maintenance. See also  datenstrom/community#167

**Reply by pftnhr** [reply]
Thank you for the prompt reply. I have installed the draft extension. I'll have a look at the code tomorrow, maybe I can fix it.

**Reply by pftnhr** [reply]
Well, I guess I’m not able to build it. But thank you anyway.

**Reply by markseuffert** [reply]
Maybe someone wants to convert [this code](https://github.com/nogginfuel/yellow-plugin-maintenance/blob/master/maintenance.php) to the new API, [make and publish](https://github.com/datenstrom/yellow-extensions/tree/master/source/publish) a maintenance extension? 

All it needs now is someone to volunteer. 😀

**Reply by GiovanniSalmeri** [reply]
If nobody else is working on this, I will do it with pleasure (also because it will be useful to me :-)

**Reply by pftnhr** [reply]
That would be awesome! Thank you in advance. I appreciate it very much.

**Reply by GiovanniSalmeri** [reply]
@pftnhr Here it is: https://github.com/GiovanniSalmeri/yellow-maintenance

I have done some quick tests, it seems to work well. I have added the possibility to whitelist IPs: this is handy for those, like me :-), who do not use the edit extension. `::1` and `127.0.0.1` (i.e. localhost) are automatically whitelisted. Comments are welcome, hope it is useful!

**Reply by pftnhr** [reply]
@GiovanniSalmeri Works very well so far. Thank you very much!
What I still don't understand is which file I have to edit or create to get a maintenance layout for the whole site.

**Reply by pftnhr** [reply]
Got it. Works as desired!

**Reply by schulle4u** [reply]
@GiovanniSalmeri thanks, looks nice and simple, will give it a test should I ever need this feature. From what I can see so far the language file entry is missing in your `extension.ini`. 

And again the general reminder / nag screen for all developers: Published extensions can be included in our update process, and will be listed in the official extension's repository. This allow users to always use the latest and greatest code, as well as easily discover new extensions. [Basically this is just one command](https://github.com/datenstrom/yellow-extensions/tree/master/source/publish) after setting up a simple development environment. 🙂

**Comment by annaesvensson** [comment]
I propose that Giovanni's [maintenance extensions](https://github.com/GiovanniSalmeri/yellow-maintenance) be made publicly available and no longer experimental. 😀

**Reply by GiovanniSalmeri** [reply]
Proposal gladly accepted! I will proceed with publication in the next few days. (I confess that I had completely forgotten that I had written this extension 😁)
